src/gen_dragonfruit/gd_dev_adafruit_MPL115A2_temp.cpp

```diff
@@ -7,10 +7,10 @@ void gd_dev_adafruit_MPL115A2_temp_open(void){
     mpl115a2t1.begin();
 }
 
-int32_t gd_dev_adafruit_MPL115A2_temp_read(void){
-  int32_t value = 0;
+uint32_t gd_dev_adafruit_MPL115A2_temp_read(void){
+  float value = 0;
   #ifndef SEN_STUB
-  value = mpl115a2t1.getTemperature() + 273.15; //Function gives C, return K
+  value = mpl115a2t1.getTemperature() + 273.15; //Function gives Celcius, return Kelvin
   #endif
-  return value;
+  return (uint32_t)value;
```

Comment:
We're still losing precision here - everything after the decimal is dropped in the conversion from a float to a uint32_t.